### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by fkaercher**

### CODE REVIEW

File: src/main/groovy/de/dkfz/eilslabs/tools/constants/Constants.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/eilslabs/tools/constants/Constants.java]: This file must stay within the Roddy core project! It is full of application specific constants.

File: src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionLevel.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionLevel.java]: This file directly relates to ExecutionContext in the Roddy core project. Please move it back.

**Review by dankwart-de — APPROVED** [review]

File: src/main/groovy/de/dkfz/eilslabs/tools/conversion/BufferValue.groovy [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public String toString(BufferUnit unit = BufferUnit.M) {
+        if (alignedValue * baseUnit.multiplier < unit.multiplier) {
+            return "" + alignedValue + baseUnit.name();
+        }
+        return "" + (alignedValue * baseUnit.multiplier / unit.multiplier as Long) + unit.name()
+    }
+
+    public Long toNumber(BufferUnit unit = BufferUnit.M) {
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/eilslabs/tools/conversion/BufferValue.groovy]: toLong()?

File: src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+package de.dkfz.eilslabs.tools.conversion;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Provides basic fields and methods for classes which represent a jobState of i.e. data sets which for a given point in time.
+ */
+public class InfoObject {
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java:15]: Do we want @CompileStatic here?

File: src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionHelper.groovy [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        Process process = command.execute();
+
+        final String separator = System.getProperty("line.separator");
+        process.waitFor();
+        if(process.exitValue()) {
+            throw new RuntimeException("Process could not be run" + separator + "\tCommand: sh -c "+ command + separator + "\treturn code is: " + process.exitValue())
+        }
+
+        def text = process.text
+        return text.length() > 0 ? text[0 .. -2] : text; //Cut off trailing "\n"
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionHelper.groovy]: Should that not be 
```
public static chomp(String text) {
  text.length() >= 2 ? text[0..-2] : text
}
```

I.e. a different condition (and maybe a dedicated function).

File: src/main/groovy/de/dkfz/eilslabs/tools/logging/IOHelperMethods.groovy [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+
+package de.dkfz.eilslabs.tools.logging
+
+
+import de.dkfz.eilslabs.tools.constants.StringConstants
+import de.dkfz.eilslabs.tools.execution.ExecutionHelper
+import groovy.io.FileType
+import org.apache.commons.codec.digest.DigestUtils
+
+//import de.dkfz.roddy.core.RunService
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/eilslabs/tools/logging/IOHelperMethods.groovy]: remove comment

**Review by vinjana — COMMENTED** [review]
Only minor things. Would be nice to have the changed or suggestions to be considered.

File: src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java (hunk 4 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java:15]: It is java code

File: src/main/groovy/de/dkfz/eilslabs/tools/conversion/BufferValue.groovy (hunk 3 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/eilslabs/tools/conversion/BufferValue.groovy]: toLong sounds good. The initial design was going for any number (which is a long at the end).

File: src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionHelper.groovy (hunk 5 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/eilslabs/tools/execution/ExecutionHelper.groovy]: Sounds good, try it.

File: src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java (hunk 4 above)
Comment (vinjana) [reply src/main/groovy/de/dkfz/eilslabs/tools/conversion/InfoObject.java:15]: ups

**Review by vinjana — APPROVED** [review]

**Pull request merged**